docs/DeveloperGuide.adoc

```diff
@@ -277,89 +301,360 @@ Refer to the guide <<DevOps#, here>>.
+|* |computing student |data saved by the app to be easily parsable |create alternative clients
+|* |infosec student |coupon data to be encrypted |hackers cannot hack my coupons
+|====================================================================================================================================================================
 
-|`* * *` |user |add a new person |
+[appendix]
+== Use Cases
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+This is a list of Use-Cases for STASH, a coupon stash application.
+Primary actor is the user.
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+.Use Cases Overview for STASH
+image::usecase.png[Use Case Diagram,500,200]
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+=== **Use Case: UC1 -  Add Coupon**
+**Actor**: `user`
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
+This use case describes how a user uses STASH to add a new coupon entry.
 
-_{More to be added}_
+**MSS**
 
-[appendix]
-== Use Cases
+. User keys in command to add a coupon entry.
+. STASH informs the user that the entry is added.
++
+Use case ends.
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+**Extensions**
 
-[discrete]
-=== Use case: Delete person
+[none]
+* 1a. STASH detects an error in the entered data.
+[none]
+** 1a1. STASH requests the user to enter the details with the correct format.
+** 1a2. User enters new data.
+** Steps 1a1 - 1a2 are repeated until the data entered are correct.
+** Use case resumes from step 2.
+
+=== **Use Case: UC2 - Listing all the coupon entries**
+**Actor**: `user`
+
+This use case describes how a user uses STASH to list out all the coupon entries.
+
+**MSS**
+
+. User keys in command to add a coupon entry.
+. STASH lists out all coupons.
++
+Use case ends.
+
+**Extensions**
+
+[none]
+* 1a. STASH detects an error in the entered data.
+[none]
+** 1a1. STASH requests the user to enter the details with the correct format.
+** 1a2. User enters new data.
+** Steps 1a1 - 1a2 are repeated until the data entered are correct.
+** Use case resumes from step 2.
+
+[none]
+* 1b. STASH detects that the data list is empty.
+[none]
+** 1b1. STASH informs the user that the list is empty.
+** Use case ends
+
+
+=== **_Use Case: UC3 - Mark a coupon as used_**
+**Actor**: `user`
+
+This use case describes how a user uses STASH to mark a coupon entry as used.
+
+**MSS**
+
+. User keys in command to mark a coupon as used.
+. STASH marks the coupon as used.
++
+Use case ends.
+
+**Extensions**
+
+[none]
+* 1a. STASH detects an error in the entered data.
```

Comment:
why is the error detected when the coupon is set to 'used' instead of when the coupon is added?